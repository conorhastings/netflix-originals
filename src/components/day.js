import React from 'react';
import { css } from 'emotion';
import Launches from './launches';

const WEEK_DAYS = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa' ];

const dayStyle = css({ 
  padding: '5%',
  borderLeft: '1px solid black',
  overflow: 'hidden'
});

const dayNumberStyle = css({
  fontWeight: '700',
  fontSize: '14px',
  lineHeight: '16px'
});

const emptyDayStyle = css(dayStyle, { backgroundColor: 'lightgray' });


const daysHeader = css({ 
  display:'grid',
  gridTemplateColumns: 'repeat(7, 1fr)',
  borderTop: '1px solid black',
  borderBottom: '1px solid black',
  borderRight: '1px solid black',
  height: '8vh', 
  fontWeight: 700
});

export function Day({ day, launches, height }) {
  return (
    <div className={dayStyle}>
      <div className={dayNumberStyle}>{day}</div>
      {
        launches
        ?
        <Launches day={day} launches={launches} dayHeight={height} />
        :
        null
      }
    </div>
  );  
}

export function EmptyDay() {
 return <div className={emptyDayStyle} />; 
}

export class DayHeader extends React.PureComponent {
  render() {
    return (
      <div className={daysHeader}>
      {
        WEEK_DAYS.map(day => <div key={day} className={dayStyle}>{day}</div>)
      }
      </div>
    );
  }
}
