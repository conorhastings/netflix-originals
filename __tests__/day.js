import React from 'react';
import renderer from 'react-test-renderer';
import { Day, EmptyDay, DayHeader } from '../src/components/day.js';


test('Day component renders without error', () => {
  const tree = renderer.create(
  	<Day 
  	  day={1}
  	  launches={undefined}
  	  height={77}
  	/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('EmptyDay component renders without error', () => {
  const tree = renderer.create(
  	<EmptyDay />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('DayHeader component renders without error', () => {
  const tree = renderer.create(
  	<DayHeader />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});