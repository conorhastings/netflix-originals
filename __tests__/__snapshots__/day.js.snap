// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day component renders without error 1`] = `
<div
  className="css-1vew5v7"
>
  <div
    className="css-1i9y2d3"
  >
    1
  </div>
</div>
`;

exports[`DayHeader component renders without error 1`] = `
<div
  className="css-1a7ck8"
>
  <div
    className="css-1vew5v7"
  >
    Su
  </div>
  <div
    className="css-1vew5v7"
  >
    Mo
  </div>
  <div
    className="css-1vew5v7"
  >
    Tu
  </div>
  <div
    className="css-1vew5v7"
  >
    We
  </div>
  <div
    className="css-1vew5v7"
  >
    Th
  </div>
  <div
    className="css-1vew5v7"
  >
    Fr
  </div>
  <div
    className="css-1vew5v7"
  >
    Sa
  </div>
</div>
`;

exports[`EmptyDay component renders without error 1`] = `
<div
  className="css-hmvjj6"
/>
`;
